api/services/ocean/spark/schemas/oceanSparkApplicationState.yaml

```diff
@@ -0,0 +1,31 @@
+description: >
+  The current state of the Spark application.
+
+
+  Transient states:
+
+  * `NEW`: when an application is created in the API, it is marked as `NEW`.
```

Comment:
How about the following? I added a bit of explanation as to what happens between new => submitted (mostly scheduling the pod) and submitted => running (mostly downloading the docker img). I removed references to the spark-operator, to make this simple to regular users who don't know what it is, though I'm not against it if you think it's important.

NEW: The application creation request was received. The k8s cluster must find a room for the Spark driver.

SUBMITTED: The Spark driver pod is starting up. The configured Docker image must be downloaded.

RUNNING: When the Spark driver pod starts running, the application is marked as RUNNING

COMPLETED: The application completed successfully. 

FAILED: The application exited with a non-zero exit code.

SUBMISSION_FAILED: The Spark application could not be started because the spark-submit operation failed.

Reply:
Better copy indeed! We'll just have to work on `NEW` and `SUBMITTED` because that's not exactly what happens.

`SUBMITTED` is all the time between spark-submit and `RUNNING`.
`NEW` is everything before spark-submit. So it's mostly Ocean Spark and spark-operator internals.

So in state `SUBMITTED`:
* The Spark driver pod is starting up.
* The k8s cluster must find a room for the Spark driver.
* Its Docker image must be downloaded.

I suggest:

NEW: The application creation request was received. The spark-submit operation is pending.

SUBMITTED: The spark-submit operation happened. The Spark driver pod is starting up: the k8 cluster must find room for the Spark driver and its Docker image must be downloaded.

Reply:
We had a follow-up discussion on Slack and I updated the PR. Thanks!